docs/api-provider-service.yaml

```diff
@@ -0,0 +1,506 @@
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/provider_offer'
+
+    post:
+      tags:
+        - offer
+      operationId: create_offer
+      description: >-
+        Server owners who wish to offer their servers for rent in
+        the marketplace fills out the required details described below
+        to create an offer and submit it to the provider service.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/project_offer'
+      responses:
+        '201':
+          description: Offer submitted to provider service successfully.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/provider_offer'
+        '400':
+          description: Invalid input data.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                status: 'Error'
+                message: >-
+                  Invalid or insufficient input parameters. Cannot create
+                  offer.
+
+  /offer/<offer_id>:
+    get:
+      tags:
+        - offer
+      operationId: offer_show(offer_id)
+      description: >-
+        Obtain details of a particular offer.
+        Admin user is allowed to fetch info about any offer.
+        Non admin user is allowed to fetch information
+        about only the offers that belong to its project.
+      responses:
+        '200':
+          description: >-
+            Successful operation will display the details of requested offer.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/provider_offer'
+        '400':
+          description: Invalid <offer_id>
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                status: 'Error'
+                message: 'Invalid offer id'
+        '403':
+          description: >-
+            You do not have sufficient privileges to execute this operation.
+            Non-admin or non-owner user is not allowed to see the offer info.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                status: 'Error'
+                message: >-
+                  You do not sufficient priviledges to perform this action
+        '404':
+          description: <offer_id> does not exist.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                status: 'Error'
+                message: 'offer id does not exist'
+    put:
```

Comment:
Do we really need an API call for updating the status? In what situations will this be used? Shouldn't the status be managed exclusively by the workflow?